### ISSUE TITLE
[ac_range_check] Implement RACL checking logic

### ISSUE THREAD

**Issue by Razer6** [lowRISC/opentitan#25454]
RACL is not yet upstreamed thus AC ranges don't implement the RACL checks and the RACL error logging. Implement that once RACL is upstream.

_Originally posted by @andreaskurth in https://github.com/lowRISC/opentitan/pull/25357#discussion_r1863150123_

**Comment by vogelpi** [comment]
I think both the checks and the error logging get addressed by #25906.

**Comment by andreaskurth** [comment]
I agree. Even though there are follow-up PRs (such as #26814) with extensions and fixes, I think we can close this issue. @Razer6 please reopen if you disagree.

**Comment by andreaskurth** [comment]
Reopening since there are still TODOs in the implementation:
https://github.com/lowRISC/opentitan/blob/033dee0b128c3aae16f3512b041016fccc884f02/hw/ip_templates/ac_range_check/rtl/ac_range_check.sv.tpl#L270
https://github.com/lowRISC/opentitan/blob/033dee0b128c3aae16f3512b041016fccc884f02/hw/ip_templates/ac_range_check/rtl/ac_range_check.sv.tpl#L274
